### ISSUE TITLE
App v2 with Wait error: context deadline exceeded (Client.Timeout exceeded while awaiting headers)

### ISSUE THREAD

**Issue by mitchellmaler** [rancher/terraform-provider-rancher2#508]
Sometimes when deploying a app v2 with wait=true I receive this error:

```
[ERROR] upgrading App V2: Get "https://rancher-url/k8s/clusters/c-gxfjr/v1/catalog.cattle.io.operations/cattle-monitoring-system/helm-operation-mt7rm": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

Instead of directly exiting and ending the resource wait can it retry to connect a few times or even wait and retry?

**Comment by rawmind0** [comment]
I don't think so, getting timeout. Http client timeout on waiting for app v2 requests is `10s` just to get helm operation from api. Is your Rancher overloaded?? Anyway, http client timeout has been extended to `60s`, included at PR #498

**Comment by mitchellmaler** [comment]
Yea I started digging and I think the node the rancher replica I was hitting was overloaded. Thanks for increasing the timeout though!
